qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/PhoneNumberValidationSteps.java

```diff
@@ -0,0 +1,128 @@
+    
+    private TestData.PatientInfo patient;
+    private RegistrationEditSectionPage registrationEditSectionPage;
+
+    @Before(RunTest.HOOK.SELENIUM_PHONE_NUMBER_VALIDATION)
+    public void setUp() throws Exception {
+        initiateWithLogin();
+        patient = createTestPatient();
+        new TestData.TestVisit(patient.uuid, TestData.getAVisitType(), getLocationUuid(homePage)).create();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_PHONE_NUMBER_VALIDATION)
+    public void tearDown() throws Exception {
+        deletePatient(patient);
+        quit();
+    }  
+    
+    @Given("user goes to find patient record app")
+    public void userGoesToFindPatientApp() {
+        findPatientPage = homePage.goToFindPatientRecord();
+    }
+
+    @When("user searches for patient using patient identifier")
+    public void searchForPatientUsingPatientIdentifier() {
+        findPatientPage.enterPatient(patient.identifier);
+    	findPatientPage.waitForPageToLoad();
+    }
+
+    @And("user clicks on the first patient")
+    public void userClicksOnFirstPatient() {
+        dashboardPage = findPatientPage.clickOnFirstPatient();
+    }
+
+    @Then("system loads patient dashboard page")
+    public void loadDashboardPage() {
+        assertPage(dashboardPage.waitForPage());
+    }
+    
+    @And("user clicks on show contact link")
+    public void userClickOnShowContact() {
+        dashboardPage.clickOnShowContact();
+    }
+
+    @And("user clicks on edit contact info link")
+    public void userClickOnEditContact() {
+        registrationEditSectionPage = dashboardPage.clickOnEditContact();
+    }
+
+    @And("user clicks on phone number tab")
+    public void userClicksOnPhoneNumber() {
+        registrationEditSectionPage.clickOnPhoneNumberEdit();
+    }
+
+    @And("user clears phone number")
+    public void userClearsPhoneNumber() {
+        registrationEditSectionPage.clearPhoneNumber();
+    }
+
+    @And("user enters an invalid phone number")
+    public void userEntersInvalidPhoneNumber() {
+        registrationEditSectionPage.enterPhoneNumber("abcdefg1");
+    }
+
+    @And("user clicks on confirm edit button")
+    public void userClicksOnConfirmEditButton() {
+        registrationEditSectionPage.clickOnConfirmEdit();
+    }
+
+    @Then("system returns a message validator")
+    public void validatePhoneNumber() {
+        assertTrue(registrationEditSectionPage.getInvalidPhoneNumberNotification().contains(("Must be a valid phone number (with +, -, numbers or parentheses)")));
+    }
+
+    @And("user clears the phone number input area")
+    public void userClearsThePhoneNumber() {
+        registrationEditSectionPage.clearPhoneNumber();
+    }
+
+    @And("user enters the correct phone number")
+    public void userEntersCorrectPhoneNumber() {
+        registrationEditSectionPage.enterPhoneNumber("111111111");
+    }
+
+    @And("user clicks again on confirm edit button")
+    public void clicksOnConfirmEditButton() {
```

Comment:
Why did you drop starting with `user`?

Reply:
Did you see the above?

Reply:
Responded the above.

Reply:
How did you respond?

Reply:
Responded it in this [commit](https://github.com/openmrs/openmrs-contrib-qaframework/pull/370/commits/3a793ce648f10c36f91b5aabd2836263cdffbc36)

Reply:
Did you understand my original comment?

Reply:
Ohh Resolved it again.